### PR TITLE
R3 Final Review

### PR DESCRIPTION
**Pull request by GoodluckH**
Note that some "upwraps" only exists in the code comments. They are there as a reminder of the work to be done for integrating with svc-storage.

### CODE REVIEW

**Comment by amsmith-pro** [comment]
Can be merged into main for R3 now that its dependencies, svc-storage and lib-common, have been merged

**Review by owlot — COMMENTED** [review]
## Documents
- [x] ICD
  - [x] Current interfaces (grpc & rest)
  - [x] Current relevant document links
- [x] SDD
  - [x] Description of grpc & rest handler implementations
  - [x] Current sequence diagrams where applicable
  - [x] Current relevant document links
- [x] CONOPS
  - [x] Current information
  - [x] Bibliography (if any citations)
- [x] README (see svc-cargo for example)
  - [x] Badges
  - [x] Accurate installation instructions
  - [x] Current links to other documents
  - [x] Arrow DAO links
- [x] Any instance of :construction: should genuinely be under construction

## Code (not applicable to examples/ folder)
- [x] No unhandled unwrap()s
- [x] Accurate (!) code comments on all functions
- [x] No `# Arguments` and `# Returns` in function comments
- [x] Function arguments checked where possible
- [x] Sufficient logger (info, warn, and debug) messages
- [x] Sensible Types (unsigned vs. signed)
- [x] No `#[allow(dead_code)]` without good reason
- [x] End all error, info, and debug messages with punctuation (periods, etc.)
- [x] No FIXMEs or TODOs without a projected release (`TODO R4: etc.`)

## REST API
- [x] Return codes accurately documented with openapi
- [x] Return codes make sense
- [x] Sufficiently descriptive error messages
- [x] Do NOT reflect a user's bad input back at them in replies.
    - i.e. "Warning: '{}' is not a valid entry." should instead be "Warning: invalid entry."
    - Use debug messages to see the invalid argument.

## Cargo.toml(s)
- [x] Package versions correct
- [x] No minor versions specified
- [x] Links to crates.io package, not git branch
- [x] Must have `license-file = "LICENSE"` under `[package]`

File: README.md
Comment (owlot) [inline README.md:1]: Banner should be: 

`Arrow Banner`

File: client-grpc/Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [package]
-description = "Assets Service GRPC client"
-keywords    = ["vtol", "client", "assets"] # max 5
+description = "Arrow assets service gRPC client"
+keywords    = ["vtol", "client", "grpc", "assets"] # max 5
```
Comment (owlot) [inline client-grpc/Cargo.toml]: missing "rest" keyword

File: server/src/rest/structs.rs
Comment (owlot) [inline server/src/rest/structs.rs:1]: The `Aircraft` struct has a `random` function implementation which isn't being used. This should be removed.
```
💀 Checking #[allow(dead_code)]...

Location                         | Statement                                                   
-------------------------------- | ------------------------------------------------------------
server/src/rest/structs.rs (266) | fn random  
```

**Review by owlot — COMMENTED** [review]
```
🚧 Checking TODOs and FIXMEs...

Test  | Location                         | Statement                                         
----- | -------------------------------- | --------------------------------------------------
  X   | server/src/rest/api.rs (101)     | // TODO R3: let _client_option = grpc_clients.s...
  X   | server/src/rest/api.rs (305)     | //TODO R3                                         
  X   | server/src/rest/api.rs (344)     | //TODO R3                                         
  X   | server/src/rest/api.rs (382)     | //TODO R3                                         
  X   | server/src/rest/api.rs (420)     | //TODO R3                                         
  X   | server/src/rest/api.rs (625)     | //TODO R3                                         
  X   | server/src/rest/api.rs (819)     | //TODO R3                                         
  X   | server/src/rest/api.rs (1107)    | //TODO R3                                         
  X   | server/src/rest/api.rs (1236)    | //TODO R3                                         
  X   | server/src/rest/structs.rs (225) | /// TODO R3/4: For now we can just say "Boeing"...
  ```

File: client-grpc/Cargo.toml (hunk 1 above)
Comment (amsmith-pro) [reply client-grpc/Cargo.toml]: https://github.com/Arrow-air/svc-assets/pull/20/commits/1ac58ee2fc85b10d9cb946507ca3201b8d77951a

File: server/src/rest/structs.rs
Comment (amsmith-pro) [reply server/src/rest/structs.rs:1]: https://github.com/Arrow-air/svc-assets/pull/20/commits/1ac58ee2fc85b10d9cb946507ca3201b8d77951a

File: README.md
Comment (amsmith-pro) [reply README.md:1]: https://github.com/Arrow-air/svc-assets/pull/20/commits/1ac58ee2fc85b10d9cb946507ca3201b8d77951a

**Review by owlot — DISMISSED** [review]

File: server/build.rs [hunk 2]
```diff
@@ -20,7 +21,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         .build_client(false)
         .compile(&[proto_file], &[proto_dir])?;
 
-    println!("cargo:rerun-if-changed={}", proto_file);
+    println!("(build) cargo:rerun-if-changed={}", proto_file);
```
Comment (owlot) [inline server/build.rs]: This should not be prefixed as the line will be parsed by the build process to determine the `rerun-if-changed` configuration

Comment (amsmith-pro) [reply server/build.rs]: https://github.com/Arrow-air/svc-assets/pull/20/commits/660bece8217d0ef1e6813c28c8c1496e123a197d

**Review by owlot — APPROVED** [review]

**Pull request merged**